server/src/test/java/org/elasticsearch/index/cache/bitset/BitSetFilterCacheTests.java

```diff
@@ -259,4 +267,58 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         }
     }
 
+    public void testShouldLoadRandomAccessFiltersEagerly() {
+        var values = List.of(true, false);
+        for (var hasIndexRole : values) {
+            for (var indexFastRefresh : values) {
+                for (var loadFiltersEagerly : values) {
+                    for (var isStateless : values) {
+                        if (isStateless && hasIndexRole) {
+                            assertEquals(
+                                loadFiltersEagerly && indexFastRefresh,
+                                BitsetFilterCache.shouldLoadRandomAccessFiltersEagerly(
+                                    bitsetFilterCacheSettings(isStateless, hasIndexRole, loadFiltersEagerly, indexFastRefresh)
+                                )
+                            );
+                        } else if (loadFiltersEagerly) {
+                            assertTrue(
+                                BitsetFilterCache.shouldLoadRandomAccessFiltersEagerly(
+                                    bitsetFilterCacheSettings(isStateless, hasIndexRole, loadFiltersEagerly, indexFastRefresh)
+                                )
+                            );
+                        } else {
+                            assertFalse(
+                                BitsetFilterCache.shouldLoadRandomAccessFiltersEagerly(
+                                    bitsetFilterCacheSettings(isStateless, hasIndexRole, loadFiltersEagerly, indexFastRefresh)
+                                )
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
I spent a bunch of time working on an integration test for this (see this diff: https://github.com/JVerwolf/elasticsearch/compare/bugfix/BitSetFilterCache...JVerwolf:elasticsearch:bugfix/BitSetFilterCache-IT-save2?expand=1#).

However, I ran into issues updating system indices and decided it wasn't worth the effort.